Documentation/Blazorise.Docs/Routes.razor.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Blazored.LocalStorage;
+using Blazorise.Docs.Services;
+using Blazorise.Modules;
+using Microsoft.AspNetCore.Components;
+
+namespace Blazorise.Docs;
+
+public partial class Routes
+{
+    [Inject] private ThemeService ThemeService { get; set; }
+    [Inject] private IJSUtilitiesModule JSUtilitiesModule { get; set; }
+    [Inject] ILocalStorageService LocalStorage { get; set; }
+
+    protected override void OnInitialized()
+    {
+        ThemeService.ThemeChanged += OnThemeChanged;
+
+        base.OnInitialized();
+    }
+
+    protected override async void OnAfterRender( bool firstRender )
+    {
+        if ( firstRender )
+        {
+            var theme = await LocalStorage.GetItemAsync<string>( "theme" );
+
+            if ( ThemeService.CurrentTheme != theme )
+            {
+                ThemeService.SetTheme( theme );
+            }
+        }
+
+        base.OnAfterRender( firstRender );
+    }
+
+    public void Dispose()
+    {
+        ThemeService.ThemeChanged -= OnThemeChanged;
+    }
+
+    async void OnThemeChanged( object sender, string theme )
+    {
+        if ( theme == "Dark" )
+        {
+            await JSUtilitiesModule.AddAttributeToBody( "data-bs-theme", "dark" );
+        }
+        else
+        {
+            await JSUtilitiesModule.RemoveAttributeFromBody( "data-bs-theme" );
+        }
+
+        await LocalStorage.SetItemAsync( "theme", theme );
```

Comment:
Using local storage with Blazor Server results in unintended behavior. 
Since the first "render" happens on the server, where local storage is not accessible, it will always serve the light (default) mode. Then, it checks for the "dark" mode and switches accordingly. This results in a quick "light theme blink" on refresh, which is exactly what's happening. The "quickness" of the blink depends on internet speed. You can throttle in devtools for a more noticeable effect. 

To my knowledge, the only solution is to store the "dark" mode value inside a cookie, which will be sent together with the first request. This ensures the first render happens in the correct dark/light theme.

Reply:
That's the limitation at this time. We could use cookies but the GDPR consent complicates it. SO I would just leave it as it is and go with the flicker.